### PR TITLE
GODRIVER-2650 Fix incorrect X509 certificate being used as username for authentication.

### PR DESCRIPTION
**Pull request by qingyang-hu**
GODRIVER-2650

## Summary
Extract the subject from the first certificate instead of the last.

## Background & Motivation
The Subject of the last certificate was returned as the username for X509 auth after [GODRIVER-2263](https://jira.mongodb.org/browse/GODRIVER-2263), which did not match the order of loading a PEM file with multiple certs.
As the ticket suggests, this change updates the certificate subject string if the `certDecodeBlock` is never set.

### CODE REVIEW

**Review by prestonvasquez — APPROVED** [review]
LGTM

File: mongo/options/clientoptions_test.go
Comment (benjirewis) [inline mongo/options/clientoptions_test.go:580]: Just to verify my understanding: I assume this test is trying to certify that we are taking [the first certificate](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/options/testdata/one-pk-multiple-certs.pem#L36-L56) in `one-pk-multiple-certs` instead of the second. Running the two certs in that file through a certificate decoder, I see the only difference between them is the common name (`CN`), which is "localhost" and "client-intermediate" for the first and second respectively. Are we relying on the `Username` field on `Credential` to not match in its `CN` if we take the second cert instead of the first?

**Review by benjirewis — CHANGES_REQUESTED** [review]
Looking good! 

A few requests: 
- Can we leave a comment above `addClientCertFromBytes` indicating that only the first certificate is honored in a file with multiple certs for internal documentation? 
- Can we add a similar comment to [this section](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/options/clientoptions.go#L751-L755) of the `SetTLSConfig` external documentation? 
- Once this merged, can we leave a comment on GODRIVER-2263 indicating how the initial change was slightly incorrect?

File: mongo/options/clientoptions_test.go
Comment (qingyang-hu) [reply mongo/options/clientoptions_test.go:580]: Yes for this PEM file. We rely only on the `CN` section to verify that we pick up the correct subject name.

**Review by benjirewis — APPROVED** [review]
LGTM assuming tests pass, thanks.

**Pull request merged**
